File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        int cnt = 0 ; 
        unordered_set<int>mp(banned.begin() , banned.end()) ; 
        for(int i = 1 ; i <= n ; i++){
            if(mp.find(i)==mp.end() && maxSum>=i){
                cnt++ ; 
                maxSum -= i ;
            }
        }
        return cnt ;
    }
};

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        x = [i for i in range(1,n+1)]
        x = list(set(x)-set(banned))
        x = sorted(x)
        currSum = sum(x)
        ans = len(x)
        while currSum > maxSum:
            currSum -= x.pop()
            ans -= 1
        return ans
              